Add API for booking and tests

mihvas opened the pull request:
- **Entity**:  
  - `House`  
  - `Booking`

- **Repository**:  
  - `HouseRepository`  
  - `BookingRepository`

- **Service**:  
  - `HouseService`  
  - `BookingService`

- **Controller**:  
  - `HouseController`  
  - `BookingController`

- **API  #Routes**:
  - `House` 
    - `GET /api/house/get`  
    - `POST /api/house/create`  
    - `DELETE /api/house/delete` 
  - `Booking`   
    - `GET /api/booking/get`  
    - `POST /api/booking/create`  
    - `PUT /api/booking/update`  
    - `DELETE /api/booking/delete`

- **Tests**:  
  - Unit-тесты для `HouseRepository` и `BookingRepository`  
  - Интеграционные тесты для `HouseController` и `BookingController`

mihvas committed:
Add API for booking and tests

Add the ability to create and edit booking requests, add booking houses. Add tests to verify the operation of the booking API.

feip-krk commented on src/Controller/BookingController.php:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    #[Route('/get', name: 'get', methods: ['GET'])]
+    public function getAllBookings(): JsonResponse
```
Странный метод, он позволяет получать персональные данные из заявок любому пользователю АПИ

feip-krk commented on src/Controller/HouseController.php:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    #[Route('/get', name: 'get', methods: ['GET'])]
+    public function getAllHouses(): JsonResponse
```
В ТЗ был метод получения "свободных" домиков, этот возвращает все

Метода для только свободных в контроллере не вижу

feip-krk replied:
Тут  сделал читерство, но ОК

Предположим домики бронируются на "сезон" и в рамках "сезона: на них ставится галочка

mihvas committed:
Update the booking of houses

Add a field for tracking the freedom of houses, the ability to receive and book only free houses.

feip-krk commented on tests/Controller/BookingControllerTest.php:
```diff
@@ -0,0 +1,121 @@
+        $this->assertResponseFormatSame('json');
+        $client->getResponse()->getContent();
+
+        $client->request('GET', '/api/booking/get');
```
Метода больше нет, а тест остался

feip-krk commented on tests/Controller/BookingControllerTest.php:
```diff
@@ -0,0 +1,121 @@
+        $data = json_decode($content, true);
+
+        $this->assertArrayHasKey('success', $data);
+        $this->assertTrue($data['success']);
```
В этом тесте ты проверяешь, что апи "не падает". Но ты не проверяешь, что оно работает верно.

В Задании сказано: протестировать апи при помощи моков, то есть подменить ответ сервиса, который делает действие, вызываемое этим апи.

И сделать тест сценария, чтобы проверить, что после вызова апи у тебя изменился файл так, как ожидалось

feip-krk replied:
Можешь оставить эти тесты как есть, за тесты интеграции приму тот, что получает список домиков

Напиши тест сценария бронирования (+ мелкие правки) - сделаешь это - отпишись мне в ЛС. Как только смогу, запущу тесты, если все ОК - приму задание.

Тест сценария может быть таким:

- замокал ссылку на файл
- написал запрос бронирования
- прочитал файл, проверил, что запрос есть в файле

feip-krk commented on tests/Repository/HouseRepositoryTest.php:
```diff
@@ -0,0 +1,92 @@
+//    protected function tearDown(): void
+//    {
+//        unlink($this->testFile);
+//    }
```
Забыл комментарий

feip-krk commented on tests/Repository/BookingRepositoryTest.php:
```diff
@@ -0,0 +1,90 @@
+//    protected function tearDown(): void
+//    {
+//        unlink($this->testFile);
+//    }
```
Забыл комментарий

feip-krk commented on tests/Repository/BookingRepositoryTest.php:
```diff
@@ -0,0 +1,90 @@
+        $params = $this->createMock(ParameterBagInterface::class);
+        $params->method('get')
+            ->with('paths.booking_csv')
+            ->willReturn($this->filePath);
```
используешь моки - молодец!

mihvas committed:
Update the tests

Add mocks to the link to the files and tests to verify that the request is in the file.

feip-krk commented:
Твой код не работает:
![image](https://github.com/user-attachments/assets/4ce21d67-f694-492a-90a8-ea76353a034c)

![image](https://github.com/user-attachments/assets/e7a05529-9094-49bf-ab98-97a003e4658a)

Для получения этих ошибок мне потребовалось доработать проект:

![image](https://github.com/user-attachments/assets/fec747ee-1e96-43f9-a4fa-0af215e28133)
![image](https://github.com/user-attachments/assets/b1a7c0b4-d56d-4744-b5b4-3ec1278779c8)

Доработай, тестируй свой код локально

feip-krk commented:
Починила конфликты у себя, тесты падают

![image](https://github.com/user-attachments/assets/4ac20c45-30b1-4cbf-9995-fd789a59c756)

mihvas committed:
Fix the tests

Change the error detection conditions and configuration for running tests.

Faner201 commented:
Есть некоторая несостыкаовка в твоем readme, установка зависимостей через composer должна идти после развертываний docker контейнеров + нужно явно указать в каком контейнере это делать: docker exec -it php_container composer install

Faner201 commented:
А зачем тебе 2 базы данных собирать в docker-compose? Ещё нужно правильно настроить установку phpunit в контейнер, чтобы можно было использовать внутряк phpstorm а не лезть в контейнер, чтобы запустить тесты

Faner201 commented:
Зачет, но подправь что выше было сказано
